protocols/imap/src/test/java/org/apache/james/imap/processor/SetAnnotationProcessorTest.java

```diff
@@ -0,0 +1,183 @@
+        when(mockMailboxManager.hasCapability(eq(MailboxCapabilities.Annotation))).thenReturn(false);
+        assertThat(new SetAnnotationProcessor(mockNextProcessor, mockMailboxManager, mockStatusResponseFactory).getImplementedCapabilities(null)).isEmpty();
+    }
+    
+    @Test
+    public void processShouldProcessRequestAndCheckMailboxExist() throws Exception {
+        MailboxPath inbox = MailboxPath.inbox(mockMailboxSession);
+
+        when(mockMailboxManager.mailboxExists(inbox, mockMailboxSession)).thenReturn(false);
+
+        processor.process(request, mockResponder, mockImapSession);
+
+        verify(mockMailboxManager).startProcessingRequest(mockMailboxSession);
+        verify(mockMailboxManager).endProcessingRequest(mockMailboxSession);
+        verify(mockMailboxManager).mailboxExists(inbox, mockMailboxSession);
+    }
+
+    @Test
+    public void processShouldResponseNoWithFailureMailboxWhenMailboxDoesNotExist() throws Exception {
+        MailboxPath inbox = MailboxPath.inbox(mockMailboxSession);
+
+        when(mockMailboxManager.mailboxExists(inbox, mockMailboxSession)).thenReturn(false);
+
+        processor.process(request, mockResponder, mockImapSession);
+
+        verify(mockMailboxManager, never()).updateAnnotations(any(MailboxPath.class), any(MailboxSession.class), eq(Lists.newArrayList(MAILBOX_ANNOTATIONS)));
+        
+        ArgumentCaptor<HumanReadableText> captorMessage = ArgumentCaptor.forClass(HumanReadableText.class);
+        ArgumentCaptor<StatusResponse.ResponseCode> captorResponsecode = ArgumentCaptor.forClass(StatusResponse.ResponseCode.class);
+        
+        verify(mockStatusResponseFactory, times(1)).taggedNo(any(String.class), any(ImapCommand.class), captorMessage.capture(), captorResponsecode.capture());
+        assertThat(captorMessage.getAllValues()).containsOnly(HumanReadableText.FAILURE_NO_SUCH_MAILBOX);
+        assertThat(captorResponsecode.getAllValues()).containsOnly(StatusResponse.ResponseCode.tryCreate());
+    }
+
+    @Test
+    public void processShouldResponseNoWithGenericFailurWhenManagerThrowMailboxException() throws Exception {
+        Logger log = mock(Logger.class);
+        when(mockImapSession.getLog()).thenReturn(log);
+
+        MailboxPath inbox = MailboxPath.inbox(mockMailboxSession);
+
+        when(mockMailboxManager.mailboxExists(inbox, mockMailboxSession)).thenReturn(true);
+        doThrow(MailboxException.class).when(mockMailboxManager).updateAnnotations(eq(inbox), eq(mockMailboxSession), eq(Lists.newArrayList(MAILBOX_ANNOTATIONS)));
+
+        processor.process(request, mockResponder, mockImapSession);
+
+        ArgumentCaptor<HumanReadableText> captor = ArgumentCaptor.forClass(HumanReadableText.class);
+        verify(mockStatusResponseFactory, times(1)).taggedNo(any(String.class), any(ImapCommand.class), captor.capture());
+        final List<HumanReadableText> params = captor.getAllValues();
```

Comment:
I would find it short and easier  to read if : you name your captors, make them fields. Params variable do not add readability IMO .